x-pack/plugins/infra/public/containers/logs/log_position/log_position_state.ts

```diff
@@ -79,26 +119,88 @@ export const useLogPositionState: () => LogPositionStateParams & LogPositionCall
     [startKey, endKey]
   );
 
+  // Allow setting `startDate` and `endDate` separately, or together
+  const updateDateRange = useCallback(
+    (newDateRange: Partial<DateRange>) => {
+      // Prevent unnecessary re-renders
+      if (!('startDateExpression' in newDateRange) && !('endDateExpression' in newDateRange)) {
+        return;
+      }
+
+      const nextStartDateExpression =
+        newDateRange.startDateExpression || dateRange.startDateExpression;
+      const nextEndDateExpression = newDateRange.endDateExpression || dateRange.endDateExpression;
+
+      if (!isValidDatemath(nextStartDateExpression) || !isValidDatemath(nextEndDateExpression)) {
+        return;
+      }
+
+      // Dates are valid, so the function cannot return `null`
+      const nextStartTimestamp = datemathToEpochMillis(nextStartDateExpression)!;
+      const nextEndTimestamp = datemathToEpochMillis(nextEndDateExpression, 'up')!;
+
+      // Reset the target position if it doesn't fall within the new range.
+      if (
+        targetPosition &&
+        (nextStartTimestamp > targetPosition.time || nextEndTimestamp < targetPosition.time)
+      ) {
+        jumpToTargetPosition(null);
+      }
+
+      setDateRange({
+        ...newDateRange,
+        startTimestamp: nextStartTimestamp,
+        endTimestamp: nextEndTimestamp,
+        timestampsLastUpdate: Date.now(),
+      });
+    },
+    [setDateRange, dateRange, targetPosition]
+  );
+
+  // `endTimestamp` update conditions
+  useEffect(() => {
+    if (dateRange.endDateExpression !== 'now') {
+      return;
+    }
+
+    // User is close to the bottom edge of the scroll. The value of
+    // `pagesAfterEnd` changes on live stream as well.
+    if (pagesAfterEnd <= DESIRED_BUFFER_PAGES) {
+      setDateRange({
+        endTimestamp: datemathToEpochMillis(dateRange.endDateExpression, 'up')!,
+        timestampsLastUpdate: Date.now(),
+      });
+    }
+  }, [dateRange.endDateExpression, pagesAfterEnd, setDateRange]);
+
   const state = {
+    isInitialized,
     targetPosition,
-    isAutoReloading,
+    isStreaming,
     firstVisiblePosition: startKey,
     pagesBeforeStart,
     pagesAfterEnd,
     visibleMidpoint,
     visibleMidpointTime: visibleMidpoint ? visibleMidpoint.time : null,
     visibleTimeInterval,
+    ...dateRange,
   };
 
   const callbacks = {
+    initialize,
     jumpToTargetPosition,
     jumpToTargetPositionTime: useCallback(
       (time: number) => jumpToTargetPosition({ tiebreaker: 0, time }),
       [jumpToTargetPosition]
     ),
     reportVisiblePositions,
-    startLiveStreaming: useCallback(() => setIsAutoReloading(true), [setIsAutoReloading]),
-    stopLiveStreaming: useCallback(() => setIsAutoReloading(false), [setIsAutoReloading]),
+    startLiveStreaming: useCallback(() => {
+      setIsStreaming(true);
+      jumpToTargetPosition(null);
+      updateDateRange({ startDateExpression: 'now-1d', endDateExpression: 'now' });
```

Comment:
Currently the live streaming doesn't fetch new entries.

I think the `updateDateRange` has to happen on a an interval (`useInterval` from `react-use`, for example), doesn't it? It looks like `streamEntriesEffect` in the `log_entries/index.ts` drives the timestamp change right now, which sounds the wrong way around. Shouldn't updating the time range on an interval should drive the loading effect? Then that `streamEntriesEffect` wouldn't have to `await` the timeout anymore, but could just react to timestamp changes as long as the `isStreaming` prop is true.

Reply:
So I tried this locally and now I understand why you were hesitant to make such changes :see_no_evil:

So now my question becomes: If we revert that latest change about the timestamp calculation, can we commit to immediately start cleaning this up so we don't release something that puts a large burden on the Elasticsearch cluster for a simple browser scroll?

If yes, then it might be prudent to roll back to the continuous `endTimestamp` update and follow this up with a refactoring of the data flow. What do you think?

Reply:
> I think the updateDateRange has to happen on a an interval

I'm not sure. `updateDateRange` also refreshes the `startTimestamp`, shifting the range and forcing a reload (...I think, because the effect that triggers a full reload has a `isStreaming` condition somewhere).

> It looks like streamEntriesEffect in the log_entries/index.ts drives the timestamp change right now, which sounds the wrong way around. Shouldn't updating the time range on an interval should drive the loading effect?

Yep, it's not obvious. The state for the streaming is handled in `logPosition` but the behaviour is handled in `logEntries`.

I wouldn't change it now. We can address this when we redo this data flow.